### PR TITLE
Use new statusbar today endpoint

### PR DESCRIPTION
**Pull request by alanhamlett**
To reduce load on the API servers, let's use a dedicated API endpoint for status bar today stats so these requests can be cached separately from dashboard and other Summaries requests.

### CODE REVIEW

File: cmd/legacy/today/today.go [hunk 1]
```diff
@@ -59,18 +58,14 @@ func Summary(v *viper.Viper) (string, error) {
 		return "", fmt.Errorf("failed to initialize api client: %w", err)
 	}
 
-	now := time.Now()
-	todayStart := time.Date(now.Year(), now.Month(), now.Day(), 0, 0, 0, 0, now.Location())
-	todayEnd := time.Date(now.Year(), now.Month(), now.Day(), 23, 59, 59, int(time.Second-time.Nanosecond), now.Location())
-
-	summaries, err := apiClient.Summaries(todayStart, todayEnd)
+	parsed, err := apiClient.Summary()
```
Comment (gandarez) [inline cmd/legacy/today/today.go]: it seems that `summary` or `today` makes more sense here.
```suggestion
	summary, err := apiClient.Summary()
```

File: pkg/api/summary.go [hunk 2]
```diff
@@ -5,29 +5,24 @@ import (
...
 	"net/http"
-	"time"
 
 	"github.com/wakatime/wakatime-cli/pkg/summary"
 )
 
-const dateFormat = "2006-01-02"
-
-// Summaries fetches summaries for the defined date range.
+// Summary fetches code stats for Today.
```
Comment (gandarez) [inline pkg/api/summary.go]: If the endpoint has changed to `/users/current/statusbar/today` also the file might be renamed to `today.go`. What do you think @dron22?

File: pkg/api/summary.go [hunk 3]
```diff
@@ -5,29 +5,24 @@ import (
...
 //
 // ErrRequest is returned upon request failure with no received response from api.
 // ErrAuth is returned upon receiving a 401 Unauthorized api response.
 // Err is returned on any other api response related error.
-func (c *Client) Summaries(startDate, endDate time.Time) ([]summary.Summary, error) {
-	url := c.baseURL + "/users/current/summaries"
+func (c *Client) Summary() (*summary.Summary, error) {
```
Comment (gandarez) [inline pkg/api/summary.go]: Maybe get rid of `summary`? 
```suggestion
func (c *Client) Today() (*summary.Summary, error) {
```

File: pkg/api/summary.go [hunk 4]
```diff
@@ -56,59 +51,44 @@ func (c *Client) Summaries(startDate, endDate time.Time) ([]summary.Summary, err
 		))
 	}
 
-	summaries, err := ParseSummariesResponse(body)
+	summary, err := ParseSummaryResponse(body)
```
Comment (gandarez) [inline pkg/api/summary.go:54]: We are parsing the result from `/users/current/statusbar/today`.
```suggestion
	today, err := ParseTodayResponse(body)
```

File: pkg/api/summary.go [hunk 5]
```diff
@@ -56,59 +51,44 @@ func (c *Client) Summaries(startDate, endDate time.Time) ([]summary.Summary, err
...
 	}
 
-	return summaries, nil
+	return summary, nil
 }
 
-// ParseSummariesResponse parses the wakatime api response into summary.Summary.
-func ParseSummariesResponse(data []byte) ([]summary.Summary, error) {
+// ParseSummaryResponse parses the wakatime api response into summary.Summary.
+func ParseSummaryResponse(data []byte) (*summary.Summary, error) {
```
Comment (gandarez) [inline pkg/api/summary.go:63]: See my comment above.

**Review by gandarez — CHANGES_REQUESTED** [review]
I left some comments and suggestions. In general I'd replace all references to `summary` since we've replaced it by dedicated `today` endpoint and are not calling `summaries` endpoint by this PR.

File: cmd/legacy/today/today.go (hunk 1 above)
Comment (alanhamlett) [reply cmd/legacy/today/today.go]: `summary` conflicts with a module name, so I'll use `today`.

File: pkg/api/summary.go (hunk 2 above)
Comment (alanhamlett) [reply pkg/api/summary.go]: Let's leave it `summary.Today` because the response is still a `summary.Summary` struct.

**Review by gandarez — APPROVED** [review]
let's get it merged and discuss better naming of changed packages. Also would be good to squash commits in this PR.

**Pull request merged**